src/main.rs

```diff
@@ -172,23 +181,50 @@ impl App {
             println!("{}", &contents);
             return Ok(())
         }
-        let mut styles = if args.get_flag("no-color") {
+        let mut style = if args.get_flag("no-color") {
             OutputStyle::plain()
         } else {
             OutputStyle::default()
         };
         if let Some(show_ws) = args.get_one::<bool>("show-whitespace") {
             if *show_ws {
-                styles.input_whitespace = styles.input_whitespace.or(Some(styles.input));
-                styles.output_whitespace = styles.output_whitespace.or(Some(styles.output));
+                style.input_whitespace = style.input_whitespace.or(Some(style.input));
+                style.output_whitespace = style.output_whitespace.or(Some(style.output));
             } else {
-                styles.input_whitespace = None;
-                styles.output_whitespace = None;
+                style.input_whitespace = None;
+                style.output_whitespace = None;
             }
         }
         let clash: Clash = serde_json::from_str(&contents)?;
 
-        clash.pretty_print(styles)
+        // Testcases subcommand
+        if let Some(values) = args.get_many::<usize>("testcases") {
+            let testcases_to_print: Vec<usize> = values.cloned().collect();
+
+            // Random default value for "print all"
+            if testcases_to_print[0] == 999 {
```

Comment:
I don't like this use of sentinel value. Rust has sum types (`enum`) for this purpose. For example:

```rust
enum TestCaseSelection {
    All,
    Selected(Vec<TestCase>),
}
```